src/license.jl

```diff
@@ -3,114 +3,88 @@
     XPAUTH = "xpauth.xpr"
-
-    candidates = []
-
-    # user sent the complete path
-    push!(candidates, xpauth_path)
-
-    # user sent directory
-    push!(candidates, joinpath(xpauth_path, XPAUTH))
-
-    # default env (not metioned in manual)
-    if haskey(ENV, "XPAUTH_PATH")
-        xpauth_path = replace(ENV["XPAUTH_PATH"], "\"" => "")
-        push!(candidates, joinpath(xpauth_path, XPAUTH))
-    end
-
-    # default lib dir
-    if haskey(ENV, "XPRESSDIR")
-        xpressdir = replace(ENV["XPRESSDIR"], "\"" => "")
-        push!(candidates, joinpath(xpressdir, "bin", XPAUTH))
+    # Search in absolute path given by the user (assuming it was a file), and as
+    # a directory by appending XPAUTH.
+    candidates = [xpauth_path, joinpath(xpauth_path, XPAUTH)]
+    # Search in the directories of the XPAUTH_PATH and XPRESSDIR environnment
+    # variables.
+    for key in ("XPAUTH_PATH", "XPRESSDIR")
+        if haskey(ENV, key)
+            push!(candidates, joinpath(replace(ENV[key], "\"" => ""), XPAUTH))
+        end
     end
-
-    # user´s lib dir
-    push!(candidates, joinpath(dirname(dirname(libxprs)), "bin", XPAUTH))
-
-    for i in candidates
-        if isfile(i)
+    # Search in `xpress/lib/../bin/xpauth.xpr`. This is a common location on
+    # Windows.
+    push!(candidates, joinpath(dirname(libdir), "bin", XPAUTH))
+    for candidate in candidates
+        # We assume a relative root directory of the shared library. If
+        # `candidate` is an absolute path, thhen joinpath will ignore libdir and
+        # return candidate.
+        filename = joinpath(libdir, candidate)
+        # If the file exists, we assume it is a license. We don't attempt to
+        # validate the contents of the file.
+        if isfile(filename)
             if verbose && !haskey(ENV, "XPRESS_JL_NO_INFO")
-                @info("Xpress: Found license file $i")
+                @info("Xpress: Found license file $filename")
             end
-            return i
+            return filename
         end
     end
-
     return error(
-        "Could not find xpauth.xpr license file. Check XPRESSDIR or XPAUTH_PATH environment variables.",
+        "Could not find xpauth.xpr license file. Set the `XPRESSDIR` or " *
+        "`XPAUTH_PATH` environment variables.",
     )
 end
+
 """
-    userlic(; liccheck::Function = emptyliccheck, xpauth_path::String = "" )
-Performs license chhecking with `liccheck` validation function on dir `xpauth_path`
+    userlic(;
+        liccheck::Function = identity,
+        verbose::Bool = true,
+        xpauth_path::String = ""
+    )
+
+Performs license checking with `liccheck` validation function on dir
+`xpauth_path`.
 """
 function userlic(;
+    liccheck::Function = identity,
     verbose::Bool = true,
-    liccheck::Function = emptyliccheck,
     xpauth_path::String = "",
 )
-
-    # change directory to reach all libs
-    initdir = pwd()
-    if isdir(dirname(libxprs))
-        cd(dirname(libxprs))
-    end
```

Comment:
we can try removing this. We´d need to test the PR before merging

Reply:
I think I've done it correctly, but try testing it.